### PR TITLE
Support running disabled tests in TestKitGen

### PR DESCRIPTION
**Pull request by LongyuZhang**
- generate disabled.mk file to list disabled tests
- disabled tests can run by categories

[ci skip]

Signed-off-by: Longyu Zhang <longyu.zhang@IBM.com>

### CODE REVIEW

**Comment by LongyuZhang** [comment]
reviewers: @llxia @renfeiw

**Comment by renfeiw** [comment]
Please add disabled.mk to gitignore file.

File: test/TestConfig/scripts/testKitGen/makeGenTool/mkgen.pl [hunk 1]
```diff
@@ -318,12 +328,13 @@ sub handle_start {
 
 sub handle_end {
 	my ($p, $elt) = @_;
+	my $neglectedFlag;
```
Comment (renfeiw) [inline test/TestConfig/scripts/testKitGen/makeGenTool/mkgen.pl]: Move initialization of `neglectedFlag` to the file scope and use it to replace `$parseTest->{'disabled'}` in line 455.

**Comment by llxia** [comment]
- Please update the readme about this feature.
- Could we attach the content of `disabled.mk`?

File: test/docs/OpenJ9TestUserGuide.md [hunk 2]
```diff
@@ -262,6 +262,15 @@ target
     ```
     element that you want to exclude.
 
+    For tests disabeld with playlist.xml, a exclude list can be generated in 'TestConfig/disabled.mk' 
+    after running 'run_configure.mk' file under 'TestConfig' folder. These tests can be executed through
+    specifying the test target or add `disabled` in front of regular target.
```
Comment (llxia) [inline test/docs/OpenJ9TestUserGuide.md]: For tests that are disabled within playlist.xml, `disabled.mk` is generated when running `make -f run_configure.mk`. It contains a list of disabled test targets. These targets can be executed through ...

**Comment by llxia** [comment]
- eca validation failed
- For future referernce, please add `[ci skip]` in the commit msg for the changes that do not require SDK compilation.
- Please squash all commits into one.

**Comment by LongyuZhang** [comment]
> * Please update the readme about this feature.
> * Could we attach the content of `disabled.mk`?

Following is part of the disabled.mk content. Thanks.

```
disabled.extended:
	@$(MAKE) vmLifecyleTests
	@$(MAKE) cmdLineTester_shrcdbgddrext_zos
	@$(MAKE) testSoftMxRemote
	@$(MAKE) testSoftMxRemote_zlinux_64
	@$(MAKE) testSoftMxRemote_zlinux_31
	@$(MAKE) testSoftMxRemote_LP4k
	@$(MAKE) testSoftMxRemote_LP64k

.PHONY: disabled.extended

disabled.extended.external:

.PHONY: disabled.extended.external

disabled.extended.external.native:

.PHONY: disabled.extended.external.native

disabled.extended.external.regular:

.PHONY: disabled.extended.external.regular

disabled.extended.functional:
	@$(MAKE) vmLifecyleTests
	@$(MAKE) cmdLineTester_shrcdbgddrext_zos
	@$(MAKE) testSoftMxRemote
	@$(MAKE) testSoftMxRemote_zlinux_64
	@$(MAKE) testSoftMxRemote_zlinux_31
	@$(MAKE) testSoftMxRemote_LP4k
	@$(MAKE) testSoftMxRemote_LP64k
```

**Comment by llxia** [comment]
Thanks @LongyuZhang

**Comment by llxia** [comment]
Jenkins test sanity all jdk8

**Comment by llxia** [comment]
Jenkins test sanity all jdk8

**Pull request merged**